presto-hive/src/main/java/com/facebook/presto/hive/HudiDirectoryLister.java

```diff
@@ -0,0 +1,135 @@
+import static com.facebook.presto.hive.HiveSessionProperties.isHudiMetadataEnabled;
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_DATANODE_DEFAULT_PORT;
+
+public class HudiDirectoryLister
+        implements DirectoryLister
+{
+    private static final Logger log = Logger.get(HudiDirectoryLister.class);
+
+    private final HoodieTableFileSystemView fileSystemView;
+    private final HoodieTableMetaClient metaClient;
+    private final boolean metadataEnabled;
+
+    public HudiDirectoryLister(Configuration conf, ConnectorSession session, Table table)
+    {
+        log.info("Using Hudi Directory Lister.");
+        this.metadataEnabled = isHudiMetadataEnabled(session);
+        this.metaClient = HoodieTableMetaClient.builder()
+                .setConf(conf)
+                .setBasePath(table.getStorage().getLocation())
+                .build();
+        HoodieEngineContext engineContext = new HoodieLocalEngineContext(conf);
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+                .enable(metadataEnabled)
+                .build();
+        this.fileSystemView = FileSystemViewManager.createInMemoryFileSystemView(engineContext, metaClient, metadataConfig);
+    }
+
+    public Optional<HoodieTableMetaClient> getMetaClient()
+    {
+        return Optional.of(metaClient);
+    }
+
+    @Override
+    public Iterator<HiveFileInfo> list(
+            ExtendedFileSystem fileSystem,
+            Table table,
+            Path path,
+            NamenodeStats namenodeStats,
+            PathFilter pathFilter,
+            HiveDirectoryContext hiveDirectoryContext)
+    {
+        log.debug("Listing path using Hudi directory lister: %s", path.toString());
+        return new HiveFileIterator(
+                path,
+                p -> new HudiFileInfoIterator(
+                        fileSystemView,
+                        metadataEnabled ? Optional.empty() : Optional.of(fileSystem.listStatus(p)),
+                        table.getStorage().getLocation(),
+                        p),
+                namenodeStats,
+                hiveDirectoryContext.getNestedDirectoryPolicy(),
+                Optional.empty());
+    }
+
+    public static class HudiFileInfoIterator
+            implements RemoteIterator<HiveFileInfo>
+    {
+        private final Iterator<HoodieBaseFile> hoodieBaseFileIterator;
+
+        public HudiFileInfoIterator(
+                HoodieTableFileSystemView fileSystemView,
+                Optional<FileStatus[]> fileStatuses,
+                String tablePath,
+                Path directory)
+        {
+            String partition = FSUtils.getRelativePartitionPath(new Path(tablePath), directory);
+            if (fileStatuses.isPresent()) {
+                fileSystemView.addFilesToView(fileStatuses.get());
+                this.hoodieBaseFileIterator = fileSystemView.fetchLatestBaseFiles(partition).iterator();
```

Comment:
can we just call the same API here? `getLatestBaseFiles()` and simplify? i think `getLatestBaseFiles()` calls `fetchLatestBaseFiles`

Reply:
`getLatestBaseFiles()` also calls `ensurePartitionLoadedCorrectly()` which adds files to fs view. Since, in this case we have already done that, so directly calling `fetchLatestBaseFiles ()`.